src/reconcile/rec_visibility.c

```diff
@@ -910,10 +910,18 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, W
         supd_restore = F_ISSET(r, WT_REC_EVICT) &&
           (has_newer_updates || F_ISSET(S2C(session), WT_CONN_IN_MEMORY));
 
-        upd_memsize = __rec_calc_upd_memsize(onpage_upd, upd_select->tombstone, upd_memsize);
-        WT_RET(__rec_update_save(
-          session, r, ins, rip, onpage_upd, upd_select->tombstone, supd_restore, upd_memsize));
-
+        /*
+         * No need to save the update chain when it meets the following:
+         * 1. No need to restore the update chain
+         * 2. No on-disk entry exist
+         * 3. No further updates exist in the update chain to be written to the history store.
+         */
+        if (supd_restore || vpack != NULL || onpage_upd->next != NULL) {
```

Comment:
Why not implement the logic in __rec_need_save_upd?

Reply:
I tried that approach and it needs more changes than the current approach. As we need to set the update with the DS flag irrespective of whether we save the update chain except when it is globally visible. I can do that change if you find that way is good.

Reply:
Can we move the code to set the ds flag outside this if section?

Reply:
Yes, moved it outside and add the conditions inside the need_save_upd function.